api/app/src/main/kotlin/packit/service/RunnerService.kt

```diff
@@ -66,4 +71,27 @@ class BaseRunnerService(
         runInfoRepository.save(runInfo)
         return res.taskId
     }
+
+    override fun getTaskStatus(taskId: String): RunInfoDto
+    {
+        val taskStatus = orderlyRunnerClient.getTaskStatuses(listOf(taskId), true)[0]
+
+        val runInfo = runInfoRepository.findByTaskId(taskId)
```

Comment:
i would move this above `val taskStatus = orderlyRunnerClient.getTaskStatuses(listOf(taskId), true)[0]` .. so we can fail early if null and not waste a API call

Reply:
yh that makes sense!